org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1270,16 +1270,23 @@ private URI asURI(String formatterUrl) {
 		if (formatterUrl == null || formatterUrl.isBlank()) {
 			return null;
 		}
-		URI uri = null;
 		try {
-			uri = new URI(ResourceUtils.toClientUri(formatterUrl));
-		} catch (URISyntaxException e1) {
-			File file = findFile(formatterUrl);
-			if (file != null && file.isFile()) {
-				uri = file.toURI();
+			URI uri = new URI(ResourceUtils.toClientUri(formatterUrl));
+			if (!uri.isAbsolute()) {
+				return getURI(formatterUrl);
 			}
+			return uri;
+		} catch (URISyntaxException e1) {
+			return getURI(formatterUrl);
+		}
+	}
+
+	private URI getURI(String formatterUrl) {
```

Comment:
Change the parameter name `formatterUrl` to `path`?

Reply:
Fixed.